### DISCUSSION TITLE
How to customize on_attach for lsp

### DISCUSSION THREAD

**Discussion by Song-Tianxiang** [NvChad/NvChad#2484] [Q&A] [answered]
The default [`lspconfig`](https://github.com/NvChad/NvChad/blob/ff99797242f37dbc118baad3d31aa125e08da90f/lua/plugins/configs/lspconfig.lua#L10-L11) file have disabled the lsp format capability, I need it and don't want to use `null-ls` (I dont know why I need it). So, how can I enable the format capability? Not just format capability, maybe there's other things I will put on the `on_attach`
function.

I know I can rewrite the whole `lspconfig` file and dont `require` the default `lspconfig` file. But it seems not so elegant.

**Comment by siduck** [comment, marked as answer]
@Song-Tianxiang for now just rewrite  on_attach, i will add conform.nvim in v3.0 and remove the code that disables default lspformatting

**Reply by eddyekofo94** [reply]
When do you think v3.0 will be out?

**Reply by siduck** [reply]
@eddyekofo94 probably 4-5months more. btw i added     conform in example_config repo, also removed that on_attach code already
